manual/manual/html_processing/js/search.js

```diff
@@ -0,0 +1,242 @@
+// Searching the OCAML API.
+// Copyright 2019-2020 San VU NGOC
+
+// Permission to use, copy, modify, and/or distribute this software
+// for any purpose with or without fee is hereby granted, provided
+// that the above copyright notice and this permission notice appear
+// in all copies.
+
+// THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+// WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+// WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE
+// AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR
+// CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS
+// OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
+// NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+// CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+    
+// Thanks @steinuil for help on deferred loading.
+// Thanks @osener, @UnixJunkie, @Armael for very helpful suggestions
+// Thanks to all testers!
+
+const MAX_RESULTS = 20;
+const MAX_ERROR = 10;
+const DESCR_INDEX = 4; // index of HTML description in index.js
+const SIG_INDEX = 6; // index of HTML signature in index.js
+const ERR_INDEX = 8; // length of each line in index.js. This is used
+		     // for storing error, except if we don't want
+		     // description and type signature, the ERR_INDEX
+		     // becomes DESCR_INDEX.
+
+let indexState = 'NOT_LOADED';
+
+// return true if we are loading the index file
+function loadingIndex (includeDescr) {
+    switch (indexState) {
+    case 'NOT_LOADED':
+	indexState = 'LOADING';
+	
+	const script = document.createElement('script');
+	script.src = 'index.js';
+	script.addEventListener('load', () => {
+	    indexState = 'HAS_LOADED';
+	    mySearch(includeDescr);
+	});
+	document.head.appendChild(script);
+	return true;
+	
+    case 'LOADING':
+	return true;
+	
+    case 'HAS_LOADED':
+	return false;
+    }
+}
+
+// check if sub is a substring of s. The start/end of the string s are
+// marked by "^" and "$", and hence these chars can be used in sub to
+// refine the search.
+function isSubString (sub, s) {
+    // s = s.toLowerCase();
+    // sub = sub.toLowerCase();
+    // case sensitive is better for OCaml modules.
+    let ss = "^" + s + "$";
+    return (ss.indexOf(sub) !== -1);
```

Comment:
This can be simplified to `ss.includes(sub)`, isn't it?

Reply:
@Octachron many thanks for the review. I am very busy this week, I hope to be able to work on this next week-end.

Reply:
This is not an issue at all. I should have finished a full review by the end of the week. Thanks for all the work!

Reply:
> This can be simplified to `ss.includes(sub)`, isn't it?

yes thanks